File: web/src/main/java/com/example/miagendaweb/AgendaController.java
package com.example.miagendaweb;

import java.util.ArrayList;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMethod;

@CrossOrigin(origins = "*", methods = {RequestMethod.GET, RequestMethod.POST, RequestMethod.PUT, RequestMethod.DELETE})
@RestController
public class AgendaController {
    private final Agenda agenda;

    public AgendaController(Agenda agenda) {
        this.agenda = agenda;
    }

    @GetMapping("/tareas")
    public ArrayList<ArrayList<Tarea>> verTareas() {
        return agenda.obtenerSemana();
    }

    @PostMapping("/tareas")
    public void agregarTarea(@RequestBody Tarea tarea) {
        agenda.agregarTarea(tarea.getDia(), tarea.getHora(), tarea.getTexto());
    }

    @PutMapping("/tareas/{dia}/{numTarea}")
    public void modificarTarea(@PathVariable("dia") int dia, @PathVariable("numTarea") int numTarea, @RequestBody Tarea tarea) {
        agenda.modificarTarea(dia, numTarea, tarea.getHora(), tarea.getTexto());
    }

    @DeleteMapping("/tareas/{dia}/{numTarea}")
    public void eliminarTarea(@PathVariable("dia") int dia, @PathVariable("numTarea") int numTarea) {
        agenda.eliminarTarea(dia, numTarea);
    }
}

File: web/src/main/java/com/example/miagendaweb/AgendaController.kt
package com.example.miagendaweb

import org.springframework.web.bind.annotation.*
import java.text.Normalizer

fun String.sinAcentos(): String {
    val normalizado = Normalizer.normalize(this, Normalizer.Form.NFD)
    return normalizado.replace(Regex("\\p{InCombiningDiacriticalMarks}+"), "")
}

@CrossOrigin(origins = ["*"], methods = [RequestMethod.GET, RequestMethod.POST, RequestMethod.PUT, RequestMethod.DELETE])
@RestController
class AgendaController(private val agenda: Agenda) {

    @GetMapping("/tareas")
    fun verTareas(): ArrayList<ArrayList<Tarea>> = agenda.obtenerSemana()

    @PostMapping("/tareas")
    fun agregarTarea(@RequestBody tarea: Tarea) {
        agenda.agregarTarea(tarea.dia.sinAcentos(), tarea.hora, tarea.texto)
    }

    @PutMapping("/tareas/{dia}/{numTarea}")
    fun modificarTarea(
        @PathVariable("dia") dia: Int,
        @PathVariable("numTarea") numTarea: Int,
        @RequestBody tarea: Tarea
    ) {
        agenda.modificarTarea(dia, numTarea, tarea.hora, tarea.texto)
    }

    @DeleteMapping("/tareas/{dia}/{numTarea}")
    fun eliminarTarea(
        @PathVariable("dia") dia: Int,
        @PathVariable("numTarea") numTarea: Int
    ) {
        agenda.eliminarTarea(dia, numTarea)
    }
}